zerver/tests/test_events.py

```diff
@@ -4558,6 +4558,40 @@ def do_test_subscribe_events(self, include_subscribers: bool) -> None:
             ),
         )
 
+        moderators_group = NamedUserGroup.objects.get(
+            name=SystemGroups.MODERATORS,
+            is_system_group=True,
+            realm=self.user_profile.realm,
+        )
+        with self.verify_action(include_subscribers=include_subscribers, num_events=1) as events:
+            do_change_stream_group_based_setting(
+                stream,
+                "can_administer_channel_group",
+                moderators_group,
+                acting_user=self.example_user("hamlet"),
+            )
+        check_stream_update("events[0]", events[0])
+        self.assertEqual(events[0]["value"], moderators_group.id)
+
+        setting_group = self.create_or_update_anonymous_group_for_setting(
+            [self.user_profile],
+            [moderators_group],
+        )
+        with self.verify_action(include_subscribers=include_subscribers, num_events=1) as events:
+            do_change_stream_group_based_setting(
+                stream,
+                "can_administer_channel_group",
+                setting_group,
+                acting_user=self.example_user("hamlet"),
+            )
+        check_stream_update("events[0]", events[0])
+        self.assertEqual(
+            events[0]["value"],
+            AnonymousSettingGroupDict(
+                direct_members=[self.user_profile.id], direct_subgroups=[moderators_group.id]
+            ),
+        )
+
```

Comment:
This is probably better as a follow-up PR, but we will perhaps end up wanting to rework this test code using `do_change_stream_group_based_setting` to loop over settings or call a helper or something.

Reply:
Fixed in #32558 (5th commit)